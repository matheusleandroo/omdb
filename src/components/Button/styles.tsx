import styled from 'styled-components'

export const ButtonContainer = styled.button`
  width: 100%;
  border-radius: 0.75rem;
  height: 2.375rem;
  padding: 1rem 0.75rem;
  background-color: ${(props) => props.theme.primary};
  color: ${(props) => props.theme.white};
  font-weight: bold;
  font-size: 1rem;
  border: 1px solid transparent;
  display: flex;
  align-items: center;
  justify-content: center;

  transition: background-color 0.2s;

  svg {
    margin-right: 0.5rem;
  }

  &:hover {
    background: ${(props) => props.theme['primary-dark']};
  }
`
