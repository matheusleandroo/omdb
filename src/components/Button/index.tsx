import React from 'react'

import { ButtonContainer } from './styles'

interface IButton extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  title: string
}

export function Button({ title = 'Buscar', ...props }: IButton) {
  return <ButtonContainer {...props}>{title}</ButtonContainer>
}
