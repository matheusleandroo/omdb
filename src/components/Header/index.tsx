import { NavLink } from 'react-router-dom'

import { useFilters } from '../../hooks/Filters'

import { HeaderContainer } from './styles'

export function Header() {
  const { filters, updateFilters } = useFilters()
  return (
    <HeaderContainer>
      <NavLink
        to="/"
        title="OMDb"
        onClick={() =>
          updateFilters({
            ...filters,
            value: '',
            page: 1,
          })
        }
      >
        OMDb
      </NavLink>
    </HeaderContainer>
  )
}
