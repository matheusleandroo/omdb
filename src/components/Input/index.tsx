import React from 'react'
import {
  UseFormRegister,
  FieldValues,
  FieldError,
  Merge,
  FieldErrorsImpl,
} from 'react-hook-form'

import { InputContainer } from './styles'

interface IInput extends React.InputHTMLAttributes<HTMLInputElement> {
  register: UseFormRegister<FieldValues>
  error: FieldError | Merge<FieldError, FieldErrorsImpl<any>> | undefined
  id: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  label?: string
  placeholder?: string
}

export function Input({
  register,
  error,
  id,
  onChange,
  label,
  placeholder,
  ...inputProps
}: IInput) {
  return (
    <InputContainer isErrored={!!error}>
      {label && <label htmlFor={id}>{label}</label>}
      <input
        {...register(id, {
          required: true,
          onChange,
        })}
        id={id}
        placeholder={placeholder}
        {...inputProps}
      />
      {error && <span>Campo obrigat√≥tio</span>}
    </InputContainer>
  )
}
