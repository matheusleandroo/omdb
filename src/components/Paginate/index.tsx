import React from 'react'

import { Container, PaginateComponent } from './styles'

interface PaginateProps {
  pageCount: number
  onPageChange(selected: number): void
  page: number
}

export function Paginate({ page, pageCount, onPageChange }: PaginateProps) {
  return (
    <Container>
      <PaginateComponent
        forcePage={page - 1}
        pageCount={pageCount}
        pageRangeDisplayed={1}
        marginPagesDisplayed={1}
        onPageChange={(selectedItem: { selected: number }) =>
          onPageChange(selectedItem.selected + 1)
        }
        previousLabel="<"
        nextLabel=">"
        breakLabel="..."
      />
    </Container>
  )
}
