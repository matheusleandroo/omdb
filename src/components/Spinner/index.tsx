import HashLoader from 'react-spinners/HashLoader'

import { defaultTheme } from '../../styles/themes/default'
import { SpinnerContainer } from './styles'

interface ISpinner {
  isLoading: boolean
}

export function Spinner({ isLoading }: ISpinner) {
  return (
    <SpinnerContainer>
      <HashLoader loading={isLoading} color={defaultTheme.primary} />
    </SpinnerContainer>
  )
}
