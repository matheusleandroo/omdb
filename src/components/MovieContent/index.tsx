import Popup from 'reactjs-popup'
import { IMovies } from '../../interfaces/IMovies'
import { MovieContainer } from './styles'
import svgImageNotFound from '../../assets/image-not-found.svg'
import { NavLink } from 'react-router-dom'
import 'reactjs-popup/dist/index.css'
import { Button } from '../Button'

interface IMovieContent {
  movie: IMovies
}

export function MovieContent({ movie }: IMovieContent) {
  return (
    <MovieContainer>
      <Popup
        trigger={
          <img
            src={movie.Poster !== 'N/A' ? movie.Poster : svgImageNotFound}
            alt="Poster do filme"
          />
        }
        position="center center"
      >
        <div>
          <p>Title: {movie.Title}</p>
          <p>Year: {movie.Year}</p>
          <p>Type: {movie.Type}</p>
          <p>imdbID: {movie.imdbID}</p>

          <NavLink to={`/details/${movie.imdbID}`}>
            <Button title="+ info" />
          </NavLink>
        </div>
      </Popup>
    </MovieContainer>
  )
}
