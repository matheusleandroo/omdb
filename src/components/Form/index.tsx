import { useForm, SubmitHandler, FieldValues } from 'react-hook-form'
import { FiSearch } from 'react-icons/fi'

import { Button } from '../Button'
import { Input } from '../Input'

import { useFilters } from '../../hooks/Filters'

import { FormContainer } from './styles'
import { useEffect } from 'react'

interface IForm {
  onSubmit: SubmitHandler<FieldValues>
  largeTitle?: boolean
  isHome?: boolean
}

export function Form({ onSubmit, largeTitle = true, isHome = true }: IForm) {
  const { filters, updateFilters } = useFilters()

  const {
    register,
    handleSubmit,
    setFocus,
    formState: { errors },
  } = useForm<FieldValues>({
    defaultValues: {
      title: filters.value,
    },
  })

  useEffect(() => {
    setFocus('title')
  }, [setFocus])

  return (
    <FormContainer
      largeTitle={largeTitle}
      isHome={isHome}
      onSubmit={handleSubmit(onSubmit)}
    >
      <h1>Pesquise suas s√©ries e filmes favoritos</h1>
      <Input
        id="title"
        type="text"
        name="title"
        placeholder="Procure seu Filme"
        register={register}
        onChange={(e) =>
          updateFilters({
            ...filters,
            value: e.target.value,
          })
        }
        error={errors.title}
      />
      <Button title="Buscar" icon={FiSearch} />
    </FormContainer>
  )
}
