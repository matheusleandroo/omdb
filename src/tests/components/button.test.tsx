import { it, expect } from 'vitest'
import TestRenderer from 'react-test-renderer'

import { Button } from '../../components/Button'

it('should be able to render a button component', () => {
  function Component() {
    return <Button title="Test" />
  }

  const testRenderer = TestRenderer.create(<Component />)
  const testInstance = testRenderer.root

  expect(testInstance.findByType(Button).props.title).toBe('Test')
})
