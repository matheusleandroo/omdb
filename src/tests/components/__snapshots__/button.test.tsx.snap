// Vitest Snapshot v1

exports[`should be able to render a button component 1`] = `
<button
  className="sc-bcXHqe XrTGf"
>
  Test
</button>
`;
